flutter/simple_chat/lib/chat_room/chat_room_page.dart

```diff
@@ -0,0 +1,217 @@
+    final (roomId, name) =
+        ModalRoute.of(context)!.settings.arguments as (int, String);
+
+    final authCubit = context.read<AuthCubit>();
+    final myId = authCubit.state.token != null
+        ? Jwt.parseJwt(authCubit.state.token!)['user_id'] ?? 0
+        : 0;
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(name),
+        actions: [
+          IconButton(
+            icon: const Icon(Icons.info),
+            tooltip: 'Channel Info',
+            onPressed: () => _navigateToChatRoomInfoPage(
+              context,
+              roomId.toString(),
+            ),
+          ),
+        ],
+      ),
+      body: BlocListener(
+        bloc: authCubit,
+        listener: (BuildContext context, AuthState state) {
+          if (state.isUnauthenticated) {
+            Navigator.pushReplacementNamed(
+                context, SimpleChatRoutes.home.route);
+          }
+        },
+        child: BlocProvider<ChatCubit>(
+          create: (_) => ChatCubit(
+            ApiRepository.instance.chat,
+            roomId: roomId,
+            myId: myId,
+          ),
+          child: _ChatRoom(roomId),
+        ),
+      ),
+    );
+  }
+
+  void _navigateToChatRoomInfoPage(
+    BuildContext context,
+    String roomId,
+  ) =>
+      Navigator.pushNamed(
+        context,
+        SimpleChatRoutes.chatRoomInfo.route,
+        arguments: roomId,
+      );
+}
+
+class _ChatRoom extends StatefulWidget {
+  const _ChatRoom(this.roomId);
+
+  final int roomId;
+
+  @override
+  State<_ChatRoom> createState() => _ChatRoomState();
+}
+
+class _ChatRoomState extends State<_ChatRoom> {
+  List<types.Message> _messages = [];
+  late types.User _user;
+
+  late StreamSubscription<DjangoflowWebsocketState>
+      _websocketStreamSubscription;
+
+  /// It is required due to Android Emulator can't access localhost directly,
+  /// if you are running on Real Android Device then use 127.0.0.1
+  static const _webSocketUrl = 'ws://10.0.2.2:8000/ws/chat/?token=';
+
+  @override
+  void initState() {
+    super.initState();
+    _loadInitialChatData();
+    _connectToWebSocket();
+    _createStreamSubscription();
+  }
+
+  Future<void> _loadInitialChatData() async {
+    final chatCubit = context.read<ChatCubit>();
+    await chatCubit.loadData();
+
+    final user = chatCubit.state.me;
+
+    if (user != null) {
+      _user = types.User(
+        id: user.id.toString(),
+        firstName: user.firstName,
+        lastName: user.lastName,
+      );
+    }
+  }
+
+  void _connectToWebSocket() {
+    final webSocketCubit = context.read<DjangoflowWebsocketCubit>();
+    final authCubit = context.read<AuthCubit>();
+    final jwtToken = authCubit.state.token;
+    webSocketCubit.connectToUri(Uri.parse('$_webSocketUrl$jwtToken'));
+  }
+
+  void _createStreamSubscription() {
+    final webSocketCubit = context.read<DjangoflowWebsocketCubit>();
+
+    _websocketStreamSubscription = webSocketCubit.stream
```

Comment:
@oguibueno We have `DjangoflowWebsocketBlocListener` widget in `djangoflow_websocket` package. I think we can use that widget to remove this steam subscription.

Reply:
I've replaced it, thank you!

https://github.com/djangoflow/djangoflow-examples/pull/14/commits/d8a974dbd55f92f4772c999810097dbc2b813a74#diff-c04e899e69bfcac2a008065be382ea7f42b942db87d016261b6ae5a91cd05bd4R130